src/relations/db.py

```diff
@@ -91,7 +91,9 @@ def _on_relation_changed(self, event: RelationChangedEvent) -> None:
         application_relation_databag = event.relation.data[self.charm.app]
 
         # Do not allow apps requesting extensions to be installed.
-        if "extensions" in unit_relation_databag or "extensions" in application_relation_databag:
+        if "extensions" in event.relation.data[
+            event.app
+        ] or "extensions" in event.relation.data.get(event.unit, {}):
```

Comment:
The extensions is passed on with the database in the event relation data. We should also change the VM charm once this PR get accepted, since it works in the same way.

Reply:
Good catch @dragomirp!